BlueMapCore/src/main/java/de/bluecolored/bluemap/core/world/mca/chunk/Chunk_1_13.java

```diff
@@ -195,6 +204,13 @@ public int getOceanFloorY(int x, int z) {
         );
     }
 
+    @Override
+    public Optional<BlockEntity> getBlockEntity(int x, int y, int z) {
+        return blockEntities.stream()
+                .filter(it -> it.getX() == x && it.getY() == y && it.getZ() == z)
+                .findFirst();
+    }
```

Comment:
(return `@Nullable BlockEntity`, see comment above)

And we should definitely try to avoid any type of object-creation here (e.g. streams) and consider using some kind of hash-lookup or array-index to not have to loop over **all** tile-entities.
People have been filling chunks with flower-pot's already and tried to render them with bluemap so i can well imagine someone having hundreds of signs as well ^^

Maybe just splitting them into the individual chunk-sections could be a good-enough approach already 🤔

Reply:
Done, replaced with packing the coordinates into a long and using that as a key for a Map. The only way to make this faster would be to make a custom map implementation that eliminates the small impact of the boxing, or bring over one of the `Long2Object*` maps from fastutil.